oeps/oep-0019-bp-developer-documentation.rst

```diff
@@ -0,0 +1,450 @@
+  semantic markup, and multiple output formats.
+
+* A single documentation format enables consistency.  It also avoids conversion
+  to rST when we require its additional features.  Our experience has shown
+  that these two benefits of standardization keep down the overall cost of
+  documentation maintenance.
+
+* rST is the standard markup format for Python project documentation.
+
+* rST does not suffer from incompatible implementations, with different
+  features supported differently by different renderers.
+
+* GitHub renders rST natively.
+
+More details on these points are in Eric Holscher's `why rST over Markdown for
+documentation`_ blog post.
+
+.. _why rST over Markdown for documentation: http://ericholscher.com/blog/2016/mar/15/dont-use-markdown-for-technical-docs/
+
+
+Best practices
+--------------
+
+* We strongly recommend keeping line length between 80 and 120 characters. Keeping it less than 120
+  leaves room for future edits.
+
+* Consistency across rST documents is not a requirement.  However, you can use the raw version of this OEP
+  as a reference example.
+
+Learning rST
+------------
+
+* If you are familiar with Markdown:
+  
+  * Here is edX's `cheatsheet comparing rST to Markdown`_.
+
+  * Here is `another comparison of Markdown and rST`_.
+
+* See the `reStructuredText (rST) Primer`_.
+
+.. _cheatsheet comparing rST to Markdown: https://github.com/edx/mdrst/blob/master/mdrst.rst
+.. _another comparison of Markdown and rST: https://gist.github.com/dupuy/1855764
+
+
+Helpful rST-related tools
+-------------------------
+
+* `Visual Studio Code`_ works pretty well.
+
+* PyCharm_ provides an rST editor and preview panel. 
+
+* `Online reStructuredText editor`_ helps, but isn't perfect.
+
+* `Online table generator`_ (supports multiple formats).
+
+* Pandoc_ can be used to convert between many formats, like Google Doc or Markdown to rST.
+
+* Other options are listed on stackoverflow_.
+
+.. _reStructuredText (RST): http://docutils.sourceforge.net/rst.html
+.. _reStructuredText (rST) Primer: http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
+.. _Visual Studio Code: https://marketplace.visualstudio.com/items?itemName=tht13.rst-vscode
+.. _PyCharm: https://www.jetbrains.com/help/pycharm/restructured-text.html
+.. _Online reStructuredText editor: http://rst.ninjs.org/
+.. _Online table generator: http://truben.no/table/
+.. _Pandoc: https://pandoc.org/
+.. _stackoverflow: https://stackoverflow.com/a/2747041
+
+Rejected alternatives
+*********************
+
+Markdown Format
+===============
+
+We are choosing to not use `Markdown (MD)`_ for documentation. A single format enables consistency, and avoids
+conversions to rST when requiring additional features. See the `Format`_ section for more details.
+The `Learning rST`_ section has resources comparing Markdown and rST.
+
+.. _Markdown (MD): https://www.markdownguide.org/
+
+Confluence
```

Comment:
I wonder if you considered adding a section here that further clarifies expectation on WHEN and HOW we will migrate from our other documentation standards (confluence, google docs, etc).  A straw-man suggestion: any time an update to documentation is required (this can be interpreted broadly, intentionally) that update must migrate that document to the new location.

Reply:
I worry that an "migrate-on-write" strategy will just mean that we never finish, and are forever left in a bad-old-docs limbo.

Reply:
@mikedikan migration is out of the scope of this OEP and each team will decide for themselves how they would like to move their docs.